crypto/src/key/secp256k1/extended_keys.rs

```diff
@@ -163,7 +163,7 @@ impl Derivable for Secp256k1ExtendedPrivateKey {
             .into();
 
         let derivation_path = {
-            let mut child_path = self.derivation_path.as_vec().clone();
+            let mut child_path = self.derivation_path.as_slice().to_vec();
```

Comment:
Since you're consuming `self` here, why not use `into_vec()` intead of a copy of that vec through the slice?